src/core/HatAllocator.ts

```diff
@@ -37,15 +39,27 @@ export class HatAllocator {
         this.toggleDecorations
       ),
 
-      vscode.window.onDidChangeTextEditorVisibleRanges(
-        this.addDecorationsDebounced
+      // An Event which fires when the active editor has changed. Note that the event also fires when the active editor changes to undefined.
```

Comment:
Liking all the comments here; helpful

Reply:
I thought so too. Some of the names aren't that clear.